### DISCUSSION TITLE
How do I add an import statement?

### DISCUSSION THREAD

**Discussion by grosch-intl** [eslint/eslint#17038] [Q&A]
When my rule fixer runs, it's potentially adding a call to a method that they have not imported.  So I need to add something like:

```ts
import { SomeNewMethod } from '...';
```

I'm not sure how to determine what goes in the file path though.

**Comment by sam3k** [comment]
Hi @grosch-intl, I am not sure what you are trying to do. Could you provide a bit more context and a reproducible demo?

**Comment by grosch-intl** [comment]
Hi @sam3k , there are two use cases that I'm looking at, explained below.  

## Import from known package
Let's say I have a rule that goes from this:

```typescript
constructor(private readonly foo: SomeService) {}
```

to this:

```typescript
readonly #foo = inject(SomeService)
```

I need to ensure that there's an `import` of `inject` from `@angular/core`.

## Import from some location in the project
The other one is when I'm adding custom code.  I have this code in a file in all of my Angular projects:

```typescript
export const ProvideValueAccessor = (classRef: ForwardRefFn): Provider => ({
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(classRef),
    multi: true
})
```

If my custom rule finds that `NG_VALUE_ACCESSOR` is provided from an `@Component`, the rule rewrites the dictionary with a call to `ProvideValueAccessor` instead.  That means the component's file now needs to do

```typescript
import {ProvideValueAccessor} from '../../some/unknown/location'
```

**Comment by sam3k** [comment]
Hi @grosch-intl, I am not very familiar with Angular so it is hard for me to help you with this. It would help someone who might be more knowledgeable if you could provide some reproducible demos.

**Comment by sam3k** [comment]
Closing this issue since it's been a few weeks since last comment.
